integration-tests/sync-tests/sync-verifier.spec.ts

```diff
@@ -0,0 +1,96 @@
+import chai, { expect } from 'chai'
+import { Wallet, BigNumber, providers } from 'ethers'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider, verifierProvider } from '../test/shared/utils'
+import { OptimismEnv } from '../test/shared/env'
+import { DockerComposeNetwork } from '../test/shared/docker-compose'
+
+describe('Syncing a verifier', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let verifier: DockerComposeNetwork
+
+  const sequencerProvider = injectL2Context(l2Provider)
+
+  /* Helper functions */
+
+  const waitForBatchSubmission = async (
+    totalElementsBefore: BigNumber
+  ): Promise<BigNumber> => {
+    // Wait for batch submission to happen by watching the CTC
+    let totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    while (totalElementsBefore.eq(totalElementsAfter)) {
+      await sleep(500)
+      totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    }
+    return totalElementsAfter
+  }
+
+  const startAndSyncVerifier = async (sequencerBlockNumber: number) => {
+    // Bring up new verifier
+    verifier = new DockerComposeNetwork(['verifier'])
+    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })
+
+    // Wait for verifier to be looping
+    let logs = await verifier.logs()
+    while (!logs.out.includes('Starting Sequencer Loop')) {
+      await sleep(500)
+      logs = await verifier.logs()
+    }
+
+    const provider = injectL2Context(verifierProvider)
```

Comment:
The verifier provider can be decoupled and instantiated at the very top of the file just like the `sequencerProvider`

Reply:
same rationale behind the "new synced verifier" per test as described in the issue! i realize this does slow tests down, but before we have a bunch of those and if they can be parallelized, i think integration tests will still be the bottleneck here. would rather have this running and gradually iterate instead of get this final in the first round

Reply:
What I am suggesting is along the lines of:

```js
function startVerifier() {
    verifier = new DockerComposeNetwork(['verifier'])
    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })

    // Wait for verifier to be looping
    let logs = await verifier.logs()
    while (!logs.out.includes('Starting Sequencer Loop')) {
      await sleep(500)
      logs = await verifier.logs()
    }
}

function syncVerifier(provider) {
    // Wait until verifier has caught up to the sequencer
    let latestVerifierBlock = (await provider.getBlock('latest')) as any
    while (latestVerifierBlock.number < sequencerBlockNumber) {
      await sleep(500)
      latestVerifierBlock = (await provider.getBlock('latest')) as any
    }

    return provider.getBlock(sequencerBlockNumber)
}
```

You can still spin up the verifier after making transactions with this abstraction. This will make the code easier to maintain longer term because the abstractions are cleaner. You no longer need to sync the verifier when bringing it up. This abstraction split will eventually need to happen as more tests are added to this test suite